cirq-core/cirq/sim/classical_simulator.py

```diff
@@ -0,0 +1,85 @@
+from typing import Dict
+from collections import defaultdict
+from cirq.sim.simulator import SimulatesSamples
+from cirq import ops, circuits
+from cirq.circuits.circuit import AbstractCircuit
+import numpy as np
+
+
+class ClassicalSimulator(SimulatesSamples):
```

Comment:
If this is just X and CNOT gates, then this name probably is not the correct one.  In particular X + CNOT is not universal.  Maybe this should be ParitySimulator as what you have is parity and not gates?

Reply:
This simulator is supposed to solve #5822 which is where the name came from.  The idea is that it should simulate only classical states.   A fast simulator for such systems would be useful for some of the resource estimation stuff we are doing.  So, it definitely is not universal, since it does not simulate any quantum states or superposition at all.  I think ParitySimulator is confusing, since it is still measuring a classical bit string and not a parity, and that name invokes comparisons to the XX parity gates which are also not simulated here.  

The corresponding name from Q# is ToffoliSimulator, which might be a better name (if we added that gate).  We could also call it something like ClassicalStateSimulator or ClassicalOperationsSimulator to clarify the intent.

Reply:
@tanujkhattar Do you have a preference for the name of this simulator?